dataflow/src/main/java/org/checkerframework/dataflow/cfg/node/NodeVisitor.java

```diff
@@ -159,6 +159,25 @@
     // Marker nodes
     R visitMarker(MarkerNode n, P p);
 
-    // Anonymous/inner/nested class declaration within a method
+    /**
+     * Visits an anonymous/inner/nested class declaration within a method.
+     *
+     * @param classDeclarationNode the {@link ClassDeclarationNode} to be visited
+     * @param p the argument of the operation implemented by this visitor
+     * @return the return type of the operation implemented by this visitor, use Void if no return
+     *     type is needed
+     */
     R visitClassDeclaration(ClassDeclarationNode classDeclarationNode, P p);
+
+    //
+    /**
+     * Visits a non-syntactical {@link ExpressionStatementNode}, which is appended to the
+     * syntactical nodes of an expression statement.
```

Comment:
```suggestion
     * Visits an expression that is used as a statement. This node is a marker after the expression node(s).
```

Reply:
Not sure how to word this. Should something about the order in the CFG should also go into the class-doc for ExpressionStatementNode?
Please also add a comment about the new node and its location to the manual: https://github.com/eisop/checker-framework/blob/master/dataflow/manual/content.tex